### PR TITLE
feat(docs): add hello-world feature on TT

### PR DESCRIPTION
**Pull request by grdddj**
Connected with https://github.com/trezor/trezor-firmware/issues/691:
- creating a `README` file with description of a simple `hello-world` feature implemented on `TT`
- the example lives on https://github.com/trezor/trezor-firmware/tree/grdddj/docs_hello_world_feature branch - and we may want to preserve it there, so that it is easily accessible

There may be some misconceptions or inaccuracies, some review is needed. It does not even try to explain all the details - it assumes readers can read the code and understand some basic concepts.

### CODE REVIEW

**Comment by matejcik** [comment]
looking good.

please move the document to `docs/misc/`

will you be adding the "Tests" section?

**Comment by grdddj** [comment]
> will you be adding the "Tests" section?

I think I will, it should be quick. For unittests we will just need to create a function returning the greeting from `msg`, so we have something to test.

> please move the document to docs/misc/

Sure, at the same time I will force-push that hello-world branch with updated changes and testing, so we can link it there.

Just wondering about the failing `CI` at https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/pipelines/481116674 - the `common prebuild` is showing some errors with the new messages and all `T1` builds are failing because they do not understand the new messages. Should we also fix these and include it in the guide?

**Comment by matejcik** [comment]
> the `common prebuild` is showing some errors

That's because the check expects the messages to start with `Helloworld` and it's actually `HelloWorld`.
Maybe just rename the messages to `HelloSomething` to sidestep the whole problem.

> all `T1` builds are failing because they do not understand the new messages

the guide should probably mention this (add to `SKIPPED_MESSAGES` in `legacy/firmware/protob/Makefile`)

**Comment by grdddj** [comment]
> That's because the check expects the messages to start with Helloworld and it's actually HelloWorld.
Maybe just rename the file to messages-hello and messages to HelloSomething to sidestep the whole problem.

That error confused me already in the past. I will put some meaningful error message there.

> the guide should probably mention this (add to SKIPPED_MESSAGES in legacy/firmware/protob/Makefile) 

Done in [b1ed9bf](https://github.com/trezor/trezor-firmware/pull/2148/commits/b1ed9bf94eae30626d598a097e9f4f06084bc6ed), together with some other optional steps (like adding new tests into `fixtures.json`)

File: docs/misc/hello_world_feature_TT.md [hunk 1]
```diff
@@ -235,8 +249,96 @@ Hello George!
...
+Code above is using the `unittest` testing framework, however not directly from python's standard library. As these unit tests are run by `micropython`, which does not have `unittest` library, we had to create the functionality ourselves in `core/tests/unittest.py` - see `from common import *`.
+
+Current code checks one usage of `_get_text_from_msg`, the only deterministic helper function we use in our feature. One could create many test vectors trying different inputs and expecting different outputs.
+
 ### Device tests
 [docs](tests/device-tests.md)
+
+Device tests (our name for integration tests) should test the whole workflow from sending the first request into Trezor to Trezor sending the final response.
+
+`Trezorlib` is used extensively in these tests as a way to request something from Trezor and then assert the expected response (it actually uses the code we created in Part 3).
```
Comment (matejcik) [inline docs/misc/hello_world_feature_TT.md:290]: when `codified`, you should use lowercase `trezorlib` even at start of sentence

Comment (grdddj) [reply docs/misc/hello_world_feature_TT.md:290]: Is there something else worth changing? Otherwise, I could force-push rebase with this change

Comment (matejcik) [reply docs/misc/hello_world_feature_TT.md:290]: please go ahead.

**Comment by matejcik** [comment]
one more thing: please rename the helloworld branch to just `hello-world` (or push the branch to upstream under the changed name)
so that it's clearer what it is in the future.

File: docs/misc/hello_world_feature_TT.md (hunk 1 above)
Comment (grdddj) [reply docs/misc/hello_world_feature_TT.md]: Done in [8596289](https://github.com/trezor/trezor-firmware/pull/2148/commits/8596289b4eaa6debc7b31a39a21dec3413cada8c)

**Comment by grdddj** [comment]
> one more thing: please rename the helloworld branch to just `hello-world` (or push the branch to upstream under the changed name)
> so that it's clearer what it is in the future.

Hello-world branch created - https://github.com/trezor/trezor-firmware/tree/hello-world

**Review by matejcik — APPROVED** [review]

**Pull request merged**